exporters/stdout/stdoutmetric/exporter.go

```diff
@@ -90,26 +90,17 @@ func (e *exporter) Shutdown(ctx context.Context) error {
 	return ctx.Err()
 }
 
-func redactTimestamps(orig metricdata.ResourceMetrics) metricdata.ResourceMetrics {
-	rm := metricdata.ResourceMetrics{
-		Resource:     orig.Resource,
-		ScopeMetrics: make([]metricdata.ScopeMetrics, len(orig.ScopeMetrics)),
-	}
+func redactTimestamps(orig *metricdata.ResourceMetrics) {
 	for i, sm := range orig.ScopeMetrics {
-		rm.ScopeMetrics[i] = metricdata.ScopeMetrics{
-			Scope:   sm.Scope,
-			Metrics: make([]metricdata.Metrics, len(sm.Metrics)),
-		}
-		for j, m := range sm.Metrics {
-			rm.ScopeMetrics[i].Metrics[j] = metricdata.Metrics{
-				Name:        m.Name,
-				Description: m.Description,
-				Unit:        m.Unit,
-				Data:        redactAggregationTimestamps(m.Data),
-			}
+		metrics := sm.Metrics
+		for j, m := range metrics {
+			data := m.Data
+			orig.ScopeMetrics[i].Metrics[j].Data = redactAggregationTimestamps(data)
 		}
 	}
-	return rm
+	if orig.ScopeMetrics == nil {
+		orig.ScopeMetrics = []metricdata.ScopeMetrics{}
+	}
```

Comment:
We do we initialize `ScopeMetrics`?  If `redactTimestamps == false` then it would remain `nil`. Am I missing something?

```suggestion
```

Reply:
In the `exporters/stdout/stdoutmetric/example_test.go`, it expects an empty array rather than nil, the test basically fails when we return nil for `ScopeMetrics` in this case, I wanted to comply with the tests rather than changing the test.


https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/stdout/stdoutmetric/example_test.go#L328

Reply:
I suggest changing the test. I would say that this is an unexpected side-effect and unnecessary allocation.

Reply:
updated as suggested